src/build.cc

```diff
@@ -382,7 +382,7 @@ void Plan::EdgeWanted(Edge* edge) {
 Edge* Plan::FindWork() {
   if (ready_.empty())
     return NULL;
-  set<Edge*>::iterator e = ready_.begin();
+  EdgeSet::iterator e = ready_.begin();
```

Comment:
I'd also recommend that typedefs have a suffix, "_t", as a clue to the reader that the type in question is not, itself, a concrete type, but is instead an alias to something else. 

E.g. it could very well be that "EdgeSet" or "EdgePtrSet" is a completely independent class.

But the _t suffix would hint that it's probably a template type, most likely std::set, and so all of the associated functions would be available.

Reply:
Please avoid _t suffix. If you want to know if an expression is a typedef, the IDE can show you.

Reply:
What IDE?

You keep making assumptions about the capabilities of the people working with the Ninja codebase.

E.g. I don't use an IDE that provides this kind of information. My reasons for not doing so are irrelevant, but one of them is that I find intellisense functionality is typically unable to provide this information at best, or worse -- provides incorrect information. For example, I've had horrible experiences with Microsoft Visual Studio's Intellisense in past projects, which is one of the major reasons why I stopped using "fat" IDEs like MSVC altogether.

Other people, such as those who are posting patches for AIX, or various IBM related systems, may not have an IDE available at all. I wouldn't know, actually.

The google style guide makes no statements about using or not using _t. https://google.github.io/styleguide/cppguide.html

The actual C++ standard uses _t *everywhere* to indicate that things are typedefs.

I think it's strongly preferable to ensure that the person working with code knows whether they're working with a typedef, or not.

If you want to continue making assumptions about the capabilities of contributors to the Ninja codebase, then put your assumptions in the contributor guidelines.

Reply:
> The names of all types — classes, structs, **type aliases**, enums, and type template parameters — have the same naming convention. Type names should start with a capital letter and have a capital letter for each new word. No underscores.

https://google.github.io/styleguide/cppguide.html#Type_Names

Reply:
I stand corrected.

I think their advice is dangerous and misguided, but nevertheless, I failed to correctly identify that they do talk about it.